java/java.source.base/src/org/netbeans/modules/java/source/TreeShims.java

```diff
@@ -364,6 +366,31 @@
 //        }
 //        return null;
 //    }
+    public static List<DocTree> getSnippetDocTreeAttributes(DocTree node) {
+        try {
+            Class gpt = Class.forName("com.sun.source.doctree.SnippetTree"); //NOI18N
+            return isJDKVersionRelease18_Or_Above()
+                    ? (List<DocTree>) gpt.getDeclaredMethod("getAttributes").invoke(node) //NOI18N
+                    : null;
```

Comment:
performance: consider putting the `isJDKVersion..()` check outside of the `Class.forName` call. This would result in a faster path on JDKs < 18.

same for the method below

Reply:
+1. I also hope we'll be able to get rid of this test sometime soon (when we move the baseline to JDK 18).

Reply:
nb-javac based on JDK 18 has been integrated with #3729, I believe the classes can now be used without reflection/version checks.

Reply:
Yes now reflection is not needed.

Reply:
I'll raise new PR soon for removing JDK 18 related checks.

Reply:
@singh-akhilesh awesome